riocli/deployment/update.py

```diff
@@ -0,0 +1,139 @@
+    help_headers_color=Colors.YELLOW,
+    help_options_color=Colors.GREEN,
+)
+@click.option('--force', '-f', '--silent', is_flag=True, default=False,
+              help='Skip confirmation')
+@click.option('--update-all', '-a', is_flag=True, default=False,
+              help='Updates all deployments')
+@click.argument('deployment-name-or-regex', type=str, default="")
+@with_spinner(text="Updating deployment...")
+def update_deployment(force: bool, deployment_name_or_regex: str, update_all: bool, spinner=None) -> None:
+    """
+    Delete the deployment from the Platform
+    """
+    client = new_client()
+
+    if not(deployment_name_or_regex or update_all):
+        return
+
+    try:
+        deployments = fetch_deployments(client, deployment_name_or_regex, update_all)
+    except Exception as e:
+        spinner.text = click.style(
+            'Failed to update deployment(s): {}'.format(e), Colors.RED)
+        spinner.red.fail(Symbols.ERROR)
+        raise SystemExit(1) from e
+
+    if not deployments:
+        return
+
+    headers = ["name", "guid"]
+    data = [[deployment.name, deployment.deploymentId] for deployment in deployments]
+
+    spinner.write("")
+    tabulate_data(data, headers)
+
+    if not force:
+        with spinner.hidden():
+            click.confirm("Do you want to update above deployment(s)?", default=True, abort=True)
+
+    try:
+
+        for deployment in deployments:
+            dep = client.get_deployment(deployment["deploymentId"])
+            component_context = get_component_context(dep.get("componentInfo", {}))
+            payload = {
+                "service_id": dep["packageId"],
+                "plan_id": dep["planId"],
+                "deployment_id": dep["deploymentId"],
+                "context": {
+                    "component_context": component_context
+                }
+            }
+            client.update_deployment(payload)
+
+        spinner.text = click.style(
+            'Deployment(s) updated successfully!', fg=Colors.GREEN)
+        spinner.green.ok(Symbols.SUCCESS)
+
+    except Exception as e:
+        spinner.text = click.style(
+            'Failed to update deployment(s): {}'.format(e), Colors.RED)
+        spinner.red.fail(Symbols.ERROR)
+        raise SystemExit(1) from e
+
+def fetch_deployments(client: Client, deployment_name_or_regex: str, update_all: bool):
+    deployments = client.get_all_deployments(phases=[DeploymentPhaseConstants.SUCCEEDED,
+                                          DeploymentPhaseConstants.PROVISIONING])
+    result = []
+    for deployment in deployments:
+        if (update_all or deployment.name == deployment_name_or_regex or
+                (deployment_name_or_regex not in deployment.name and
+                 re.search(deployment_name_or_regex, deployment.name))):
+            result.append(deployment)
+    return result
+
+def get_component_context(component_info):
+    result = {}
+
+    for component in component_info:
+
```

Comment:
Generally, you should not leave empty lines right after you start a function block, if block, for-loop block, etc.